app/src/main/java/com/epfl/neighborfood/neighborfoodandroid/database/firebase/FirebaseDB.java

```diff
@@ -0,0 +1,69 @@
+package com.epfl.neighborfood.neighborfoodandroid.database.firebase;
+
+import android.annotation.SuppressLint;
+
+import com.epfl.neighborfood.neighborfoodandroid.database.DBData;
+import com.epfl.neighborfood.neighborfoodandroid.database.DBDataCollection;
+import com.epfl.neighborfood.neighborfoodandroid.database.DummyDatabase;
+import com.epfl.neighborfood.neighborfoodandroid.database.TentativeDB;
+import com.google.android.gms.tasks.Task;
+import com.google.android.gms.tasks.Tasks;
+import com.google.firebase.auth.FirebaseAuth;
+import com.google.firebase.firestore.FirebaseFirestore;
+
+import java.util.Map;
+
+public class FirebaseDB implements TentativeDB {
+
+    private static final String TAG = "FirebaseDB";
+    private static FirebaseDB instance;
+
+
+    private static FirebaseFirestore database;
+
+
+    public FirebaseDB() {
+        //database = FirebaseFirestore.getInstance();
+        //FirebaseAuth.getInstance();
+    }
+
+    public static  FirebaseDB getInstance() {
+        if (instance== null) {
+            instance = new FirebaseDB();
+        }
+        return instance;
+    }
+
+
+    @Override
+    public Task<DBData> fetch(String collectionPath, String documentPath) {
+        return database.collection(collectionPath).document(documentPath)
+                .get().onSuccessTask((documentSnapshot ->
+                        Tasks.forResult(new FirebaseData(documentSnapshot))
+                ));
+    }
```

Comment:
Can you explain why you are using paths for the database queries

Reply:
Response above